nipype/__init__.py

```diff
@@ -60,15 +60,55 @@ def get_info():
                          Rename, Function, Select, Merge)
 
 
-if config.getboolean('execution', 'check_version'):
+def sys_based_cache(condition):
+    def decorator(func):
+        if condition:
+            return func
```

Comment:
This means that if somebody's using Python 2, we're going to ping on every version. Since we're only worrying about one function with one boolean argument for a couple more releases, we can do a very quick-and-dirty memoization:

```Python
class _memoize(object):
    def __init__(self, func):
        self.cache = {}
        self.func = func
    def __call__(self, *args, **kwargs):
        argtuple = (args,) + tuple(kwargs.items())
        if argtuple not in self.cache:
            self.cache[argtuple] = self.func(*args, **kwargs)
        return self.cache[argtuple]
```

Another option is to manage a global state ourselves

```Python
etelemetry_results = {}

def check_latest_version(raise_exception=False):
    if raise_exception not in etelemetry_results:
        ...  # Current function
        etelemetry_results[raise_exception] = latest
    return etelemetry_results[raise_exception]
```

Or we could use something like [memoization](https://github.com/lonelyenvoy/python-memoization), which is surely a big hammer, but won't add code that will be most sensibly removed in two months.

Reply:
Sorry, thinko. That should be "ping on every interface/node/workflow instantiation", in case your brain didn't make the leap. I think it's a pretty major drawback, although I don't know how many Python 2 users we still have.